app/models/user.js

```diff
@@ -17,12 +17,35 @@ export default DS.Model.extend({
   // relationships
   token: DS.belongsTo('token', {async: true}),
   roles: DS.hasMany('role', {async:true}),
-  organizations: DS.hasMany('organization', {async:true}),
+  //organizations: DS.hasMany('organization', {async:true}),
   sshKeys: DS.hasMany('ssh-key', {async:true}),
 
   // check ability, returns a promise
   // e.g.: user.can('manage', stack).then(function(boolean){ ... });
   can: function(scope, stack){
     return can(this, scope, stack);
-  }
+  },
+
+  organizations: function() {
+    var organizations = [];
+
+    this.get('roles').forEach(function(role) {
+      var organizationId = role.get('_data.links.organization');
+      var organization = organizations.findBy('organizationId', organizationId);
+
+      if(!organization) {
+        organization = {
+          organizationId: organizationId,
+          organization: role.get('organization')
+        };
+        organizations.push(organization);
+      }
+    });
+
+    return organizations.map(function(organization) {
+      return organization.organization;
+    });
+
+  }.property('roles.[]')
```

Comment:
Thanks Matt, this is an awesome simplification.

`uniq()` doesn't seem to work though.  According to the docs:

> This only works on primitive data types, e.g. Strings, Numbers, etc.

Is there a way to get it to work with models?